velox/exec/fuzzer/FuzzerUtil.cpp

```diff
@@ -230,6 +231,85 @@ bool containsUnsupportedTypes(const TypePtr& type) {
       containsType(type, INTERVAL_DAY_TIME());
 }
 
+// Determine whether type is or contains typeName. typeName should be in lower
+// case.
+bool containTypeName(
+    const exec::TypeSignature& type,
+    const std::string& typeName) {
+  auto sanitizedTypeName = exec::sanitizeName(type.baseName());
+  if (sanitizedTypeName == typeName) {
+    return true;
+  }
+  for (const auto& parameter : type.parameters()) {
+    if (containTypeName(parameter, typeName)) {
+      return true;
+    }
+  }
+  return false;
+}
+
+bool usesTypeName(
+    const exec::FunctionSignature& signature,
+    const std::string& typeName) {
+  if (containTypeName(signature.returnType(), typeName)) {
+    return true;
+  }
+  for (const auto& argument : signature.argumentTypes()) {
+    if (containTypeName(argument, typeName)) {
+      return true;
+    }
+  }
+  return false;
+}
+
+// If 'type' is a RowType or contains RowTypes with empty field names, adds
+// default names to these fields in the RowTypes.
+TypePtr sanitize(const TypePtr& type) {
+  if (!type) {
+    return type;
+  }
+
+  switch (type->kind()) {
+    case TypeKind::ARRAY:
+      return ARRAY(sanitize(type->childAt(0)));
+    case TypeKind::MAP:
+      return MAP(sanitize(type->childAt(0)), sanitize(type->childAt(1)));
+    case TypeKind::ROW: {
+      auto children = asRowType(type)->children();
+      const auto& childNames = asRowType(type)->names();
+      std::vector<std::string> fieldNames;
+      for (auto i = 0; i < children.size(); ++i) {
```

Comment:
do the children also need to be sanitized recursively?

Reply:
Nice catch. Thanks!